.github/workflows/e2e_test_ci.yml

```diff
@@ -0,0 +1,325 @@
+name: E2E Test CI
+
+on:
+  push:
+    branches:
+      - trunk
+      - release-*
+
+  pull_request:
+    branches:
+      - trunk
+
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build ${{ matrix.name }} binaries
+    runs-on: ${{ matrix.runner }}
+    env:
+      GOVER: 1.22
+      GOOS: ${{ matrix.target_os }}
+      GOARCH: ${{ matrix.target_arch_go }}
+
+    strategy:
+      matrix:
+        include:
+          - name: Linux x64
+            runner: rust
+            target_os: linux
+            target_arch: x86_64
+            target_arch_go: amd64
+          # - name: macOS aarch64 (Apple Silicon)
+          #   runner: macos-14
+          #   target_os: darwin
+          #   target_arch: aarch64
+          #   target_arch_go: arm64
+          # - name: macOS x64 (Intel)
+          #   runner: rust-osx-x64
+          #   target_os: darwin
+          #   target_arch: x86_64
+          #   target_arch_go: amd64
+          # - name: Windows x64
+          #   runner: rust-windows-x64
+          #   target_os: windows
+          #   target_arch: x86_64
+          #   target_arch_go: amd64
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set REL_VERSION from version.txt
+        run: python3 ./.github/scripts/get_release_version.py
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GOVER }}
+
+      - name: Set up Rust
+        uses: ./.github/actions/setup-rust
+        with:
+          os: ${{ matrix.target_os }}
+
+      - name: Restore build cache (macOS)
+        if: matrix.target_os == 'darwin'
+        run: |
+          mkdir -p target
+          if [ -d /Users/spiceai/build/target ]; then
+            rsync -av /Users/spiceai/build/target/ target/
```

Comment:
Should we use the GH Cache action for these?

Reply:
That step is executed only only on our `rust` runner which persists cache, so for this case it is faster than GH Cache